### ISSUE TITLE
[Bug] tune.run throws error 

### ISSUE THREAD

**Issue by grechaw** [ray-project/ray#20253]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

This is an escalation from a customer issue.

Ray 1.7.0, running from Ray Client.
After a tune job starts and runs for a few minutes, it fails with the following stack trace:

```
types.RayTaskError(TypeError): ray::run() (pid=451, ip=172.30.5.7)

  File "/home/ray/anaconda3/lib/python3.8/site-packages/ray/tune/tune.py", line 588, in run

    runner.step()

  File "/home/ray/anaconda3/lib/python3.8/site-packages/ray/tune/trial_runner.py", line 583, in step

    next_trial = self._get_next_trial()  # blocking

  File "/home/ray/anaconda3/lib/python3.8/site-packages/ray/tune/trial_runner.py", line 727, in _get_next_trial

    self._update_trial_queue(blocking=wait_for_trial)

  File "/home/ray/anaconda3/lib/python3.8/site-packages/ray/tune/trial_runner.py", line 1190, in _update_trial_queue

    trial = self._search_alg.next_trial()

  File "/home/ray/anaconda3/lib/python3.8/site-packages/ray/tune/suggest/basic_variant.py", line 318, in next_trial

    if self.max_concurrent > 0 and len(

TypeError: '>' not supported between instances of 'list' and 'int'

2021-11-10 16:16:20.877 | ERROR    | __main__:tune:361 - Tuning task didn't return a result

2021-11-10 16:16:20.878 | DEBUG    | __main__:tune:356 - Awaiting tune task 2

2021-11-10 16:16:20.878 | ERROR    | __main__:tune_model:285 - Error in executing Ray Tune for HOLTWINTERS

2021-11-10 16:16:20.878 | ERROR    | __main__:tune_model:286 - 'NoneType' object has no attribute 'get_best_trial'

Traceback (most recent call last):
```

### Versions / Dependencies

Python 3.8
Client and Server (Anyscale) running Ray 1.7.0

### Reproduction script

I am working to get a reproduction script from the customer.

### Anything else

This issue is blocking our customer currently and we're working to resolve.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by krfricke** [comment]
There is a mismatch of arguments when using Ray client:

https://github.com/ray-project/ray/blob/releases/1.7.0/python/ray/tune/tune.py#L351-L353
vs
https://github.com/ray-project/ray/blob/releases/1.7.0/python/ray/tune/tune.py#L104-L107

We fixed this in 1.8.0. I'll confirm if there's a workaround if we just remove deprecated arguments from the call to  `tune.run`

**Comment by krfricke** [comment]
Closing because bug is already fixed - but will add the relevant PR to the 1.7.x patch release milestone
